scripts/test_hardening.js

```diff
@@ -0,0 +1,50 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+var execFileSync = require('child_process').execFileSync;
+var readdirSync = require('fs').readdirSync;
+var path = require('path');
+var glob = require('glob');
+var program = require('commander');
+
+program
+  .name('node scripts/test_hardening.js')
+  .arguments('[file...]')
+  .description(
+    'Run the tests in test/harden directory. If no files are provided, all files within the directory will be run.'
+  )
+  .action(function(files) {
+    if (files.length === 0) files = allFiles();
+    files.forEach(function(file) {
+      var files = glob.sync(file);
```

Comment:
I actually see it as a kind of feature of the language - similar to how I in the child function redefine `file` in the function arguments section. But I can rename it if you think it's confusing for the reader.

I'll refactor away `allFiles` so that `glob.sync` is always used to walk the directory.